src/discordtipbot/discordbot.cr

```diff
@@ -533,6 +535,46 @@ class DiscordBot
     end
   end
 
+  def grouplucky(msg : Discord::Message, cmd_string : String)
+    return reply(msg, "**ERROR**: This command doesn't work in DMs") if private?(msg)
+
+    cmd_usage = "#{@config.prefix}lucky [amount]"
+
+    # cmd[0]: command, cmd[1]: amount"
+    cmd = cmd_string.split(" ")
+
+    return reply(msg, cmd_usage) unless cmd.size > 1
+
+    amount = amount(msg, cmd[1])
+    return reply(msg, "**ERROR**: You have to specify an amount! #{cmd_usage}") unless amount
+
+    people = amount(msg, cmd[2]) # number of people to get lucky.
```

Comment:
This doesn't look quite right!

Reply:
The code its self or the comment, re writing the comment

Reply:
Why are you parsing the people count with the `amount` function which exists to parse a string to a BigDecimal? I think what you are looking for is `cmd[2].to_i32`

Reply:
that's unsafe if `cmd[2]` is an invalid int32

Reply:
What if I were to put a `return reply unless` after `cmd[2].to_int32`?

Reply:
Is there actually a safe way? Probably just being stupid right now...

Reply:
[`String#to_i?`](https://crystal-lang.org/api/0.24.1/String.html#to_i%3F%28base%3AInt%3D10%2Cwhitespace%3Dtrue%2Cunderscore%3Dfalse%2Cprefix%3Dfalse%2Cstrict%3Dtrue%29-instance-method) and handling the value

Reply:
Like this:
```
people = cmd[2].to_int32 # number of people to give money to.
return reply(msg, "**ERROR**: You have to specify a number of people.") unless people
```

Reply:
Yes

Reply:
A more specific error message should be given. You're checking if the value is a *valid number*, not whether the value is absent